### PR TITLE
Multiple bands

### PR DESCRIPTION
**Pull request by virginiayung**
First attempt at dealing with ingesting + stacking multiple bands as discussed in https://github.com/mapbox/rio-toa/issues/8

### CODE REVIEW

File: rio_toa/reflectance.py [hunk 1]
```diff
@@ -84,37 +115,72 @@ def calculate_landsat_reflectance(src_path, src_mtl, dst_path, creation_options,
...
+        'date_collected': date_collected,
+        'time_collected_utc': time_collected_utc,
+        'stack': stack,
+        'bands': len(bands)
     }
 
-    with riomucho.RioMucho([src_path], dst_path, _reflectance_worker,
-        options=dst_profile,
-        global_args=global_args) as rm:
+    if stack:
```
Comment (dnomadb) [inline rio_toa/reflectance.py]: @virginiayung shouldn't we just:
- stack if the number of inputs > 1
- not stash if it is equal to 1 ?

I can't think of a case where we'd want multiple inputs AND multiple outputs

File: rio_toa/reflectance.py [hunk 2]
```diff
@@ -84,37 +115,72 @@ def calculate_landsat_reflectance(src_path, src_mtl, dst_path, creation_options,
...
+    date_collected = mtl['L1_METADATA_FILE']['PRODUCT_METADATA']['DATE_ACQUIRED']
+    time_collected_utc = mtl['L1_METADATA_FILE']['PRODUCT_METADATA']['SCENE_CENTER_TIME']
 
     dst_dtype = np.__dict__[dst_dtype]
+    M, A = [], []
 
-    with rio.open(src_path) as src:
-        dst_profile = src.profile.copy()
+    for band, src_path in zip(bands, src_paths):
+        M.append(toa_utils._load_mtl_key(mtl,
```
Comment (dnomadb) [inline rio_toa/reflectance.py]: I know I wrote this function but we should go back to strict dictionary usage:

```
M = [
    toa['L1_METADATA_FILE']['RADIOMETRIC_RESCALING']['REFLECTANCE_MULT_BAND_{}'.format(b)]
        for b in bands
]
```

File: rio_toa/reflectance.py [hunk 3]
```diff
@@ -45,32 +53,61 @@ def reflectance(img, MR, AR, E, src_nodata=0):
...
     TODO
     ----
     integrate rescaling functionality for
     different output datatypes
     """
+    data = riomucho.utils.array_stack(
+            [src.read(window=window).astype(np.float32)
+                for src in open_files])
+
+    M_stack = np.ones(data.shape)
```
Comment (dnomadb) [inline rio_toa/reflectance.py]: @virginiayung I don't think we need to make a whole array for these variables - a single value for depth axis should suffice?

**Comment by virginiayung** [comment]
Merging into `the-sunangle-also-rises`

**Pull request merged**